buildlib/azure-pipelines.yml

```diff
@@ -17,6 +17,28 @@ resources:
       endpoint: ucfconsort_registry
 
 stages:
+  - stage: Codestyle
+    jobs:
+      # Check that commit title matches code style guidelines
+      - job: commit_title
+        displayName: commit title
+        steps:
+          - bash: |
+              set -eE
+              range="remotes/origin/$(System.PullRequest.TargetBranch)..$(Build.SourceVersion)"
+              for sha1 in `git log $range --format="%h"`
+              do
+                  title=`git log -1 --format="%s" $sha1`
+                  if echo $title | grep -qP '^Merge |^[A-Z/_]*: \w'
+                  then
+                      echo "Good commit title: '$title'"
+                  else
+                      echo "Bad commit title: '$title'"
+                      echo "See https://github.com/openucx/ucx/wiki/Guidance-for-contributors#general-guidelines"
+                      exit 1
```

Comment:
1. For PRs with multiple commits, it might be a good idea to check all titles.
2. You can also reuse [this](https://github.com/openucx/ucx/blob/master/buildlib/azure-pipelines.yml#L62-L63) error reporting, so AZP will report words and not `Bash exited with code '1'.` message.

Reply:
1. This is what is PR is doing - goes over a range of commits
2. 10x will add it

Reply:
1. Yeah, but it stops on the first bad one. What if there are more such bad titles?

Reply:
@amaslenn how about this : https://github.com/openucx/ucx/pull/4402/checks?check_run_id=292942883